src/lib/logical_query_plan/mock_node.cpp

```diff
@@ -103,11 +97,22 @@ void MockNode::set_table_statistics(const std::shared_ptr<TableStatistics>& tabl
   _table_statistics = table_statistics;
 }
 
-void MockNode::set_soft_key_constraints(const TableKeyConstraints& key_constraints) {
+void MockNode::set_key_constraints(const TableKeyConstraints& key_constraints) {
   _table_key_constraints = key_constraints;
 }
 
-const TableKeyConstraints& MockNode::soft_key_constraints() const { return _table_key_constraints; }
+const TableKeyConstraints& MockNode::key_constraints() const { return _table_key_constraints; }
+
+void MockNode::set_functional_dependencies(const std::vector<FunctionalDependency>& fds) {
+  _functional_dependencies = fds;
+}
+
+std::vector<FunctionalDependency> MockNode::functional_dependencies() const {
+  Assert(_table_key_constraints.empty() || !_functional_dependencies.empty(),
```

Comment:
Why is `!_table_key_constraints.empty() && !_functional_dependencies.empty()` a valid case. Shouldn't we avoid having potentially mismatching constraints and dependencies?

Reply:
I thought of two cases:

* Whenever FDs are set manually (which is only possible for MockNodes), one seems to be aware of the different logic to StoredTableNodes.

* No key constraints being set means no resulting FDs in StoredTableNodes. Same for MockNode.

Reply:
`!_table_key_constraints.empty() && !_functional_dependencies.empty()` being true means, the FDs have been set manually before.